torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -29,6 +30,40 @@
 )
 from torch.distributed.utils import _to_kwargs
 
+RESHARD_AFTER_FORWARD_STRATEGIES = {
+    HandleShardingStrategy.FULL_SHARD,
+    HandleShardingStrategy.HYBRID_SHARD,
+}
+
+@no_type_check
+def _validate_hybrid_shard_setup(state: _FSDPState, fsdp_module: nn.Module):
+    """
+    Performs validation that hybrid sharding strategy is setup. In particular, we:
+    1) Ensure root and passed in FSDP module have the same hybrid sharding strategy,
+    i.e. both should be using the same hybrid shard strategy or no hybrid shard at all.
+    2) Ensure that inter and intra-node process groups are the same across root and
+    this FSDP module.
+    """
+    if (
+        state.sharding_strategy in HYBRID_SHARDING_STRATEGIES
+        or fsdp_module.sharding_strategy in HYBRID_SHARDING_STRATEGIES
+    ):
+        if state.sharding_strategy != fsdp_module.sharding_strategy:
+            raise ValueError(
+                "When using hybrid sharding strategy, expect sharding strategies"
+                f" to be the same, but got {state.sharding_strategy} vs {fsdp_module.sharding_strategy}"
+            )
+
+        # Ensure inter and intra-node process groups are the same
+        # TODO (rohan-varma) unclear whether these should be asserts or Exceptions
```

Comment:
Are we intending to resolve this TODO before landing?

My opinion is to make these exceptions and make our own FSDP code solid to ensure an incorrect configuration cannot happen.

Reply:
[Not blocking] However, I do not see why not address this in this PR before landing.